persist/sqlite/scan_test.go

```diff
@@ -302,4 +303,123 @@ func TestScan(t *testing.T) {
 			log.Fatal("SectorSize = 0 on host that's supposed to be active")
 		}
 	}
+
+	{
+		dbHosts, err := e.QueryHosts(explorer.HostQuery{V2: true}, explorer.HostSortAcceptingContracts, explorer.HostSortAsc, 0, math.MaxInt64)
```

Comment:
I would extract `QueryHosts` into its own tests and extend it using a table test that verifies every query parameter. We have a lot of sort columns and some of these deal with currencies, so especially for these we have to assert we're performing a numerical sort.